supervisor/backups/backup.py

```diff
@@ -433,7 +433,13 @@ def _validate_file() -> bool:
                     _LOGGER.exception("Unexpected error validating password")
                     return True
 
-        return await self.sys_run_in_executor(_validate_file)
+        try:
+            await self.sys_run_in_executor(_validate_file)
```

Comment:
The return statement here was actually missing :see_no_evil: 

I've added a test to cover `validate_file`, with that this should no longer happen.